### PR TITLE
Pagination test

### PR DESCRIPTION
**Pull request by DevJS**
- updated reducer to handle BASEENTITY_LINK_CHANGE
- added pagination handling 
- added speeds up from @zcallan and @matthaywardwebdesign 
- added loading indicator from @matthaywardwebdesign 
- added query cache from @matthaywardwebdesign

### CODE REVIEW

File: src/Root.js [hunk 1]
```diff
@@ -10,12 +11,12 @@ import './utils/layouts-dev';
...
-/* --- ADDED FOR LOCAL LAYOUT DEV --- */
-// if ( !this.f ) {
-//   this.f = true;
-//   global.LayoutsDev.load( 'PROJECT_HERE' );
-// }
-/* ------------------------ */
+// // /* --- ADDED FOR LOCAL LAYOUT DEV --- */
+if ( !this.f ) {
+  this.f = true;
+  global.LayoutsDev.load( 'internmatch-new' );
```
Comment (matthaywardwebdesign) [inline src/Root.js]: We need to comment this out again.

File: src/config/config.dev.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ export default {
     },
   },
   signature: {
-    url: 'https://signatures.outcome-hub.com/signature',
+    url: 'https://signatures.outcome-hub.com',
```
Comment (matthaywardwebdesign) [inline src/config/config.dev.js]: Is this correct? I remember it having to be /signature?

File: src/Root.js (hunk 1 above)
Comment (DevJS) [reply src/Root.js]: dammit.

File: src/config/config.prod.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ export default {
     },
   },
   signature: {
-    url: 'https://signatures.outcome-hub.com/signature',
+    url: 'https://signatures.outcome-hub.com',
```
Comment (matthaywardwebdesign) [inline src/config/config.prod.js]: Is this correct? I remember it having to be /signature?

Comment (DevJS) [reply src/config/config.prod.js]: Changed!

File: src/Root.js (hunk 1 above)
Comment (DevJS) [reply src/Root.js]: Could we write a pre-commit to check on that? I do that all the time, same as Nischal and everyone else actually!

Comment (matthaywardwebdesign) [reply src/Root.js]: We should do that yeah! Just do like a grep for an uncommented global.LayoutsDev. @zcallan

File: src/config/config.prod.js (hunk 3 above)
Comment (matthaywardwebdesign) [reply src/config/config.prod.js]: Double check with @ngprnk, he may have changed the service.

**Pull request merged**